### PR TITLE
Add support for 'loading' unit tests, which evaluate a LOADING phase.

### PR DESCRIPTION
**Pull request by kkress**
This is a relatively simple addition to unittest that statically creates
rules that either explicitly fail or not depending on if the test case is valid during LOADING phase of bazel.  The test conditions are evaluated entirely in loading phase, but if we want an actual test to fail rather than just `fail()` killing the build, we can use this to assert state and report test failures.

### CODE REVIEW

File: lib/unittest.bzl [hunk 1]
```diff
@@ -564,6 +564,78 @@ def _target_under_test(env):
...
+def _loading_suite(env):
+    """Creates a test suite for loading phase tests.
+
+    Args:
+       env: env created with loadingtest.make
+
+    Returns:
+       None, creates test_suite
+    """
+    pass
```
Comment (tetromino) [inline lib/unittest.bzl:614]: This function is no-op. Was it left over from a previous iteration of the PR?

File: lib/unittest.bzl [hunk 2]
```diff
@@ -590,3 +662,9 @@ analysistest = struct(
     target_bin_dir_path = _target_bin_dir_path,
     target_under_test = _target_under_test,
 )
+
+loadingtest = struct(
+    make = _loading_make,
+    asserts = _loading_asserts,
```
Comment (tetromino) [inline lib/unittest.bzl:668]: Nit: please rename to "assert", not "asserts" for consistency with the `asserts` struct defined above (in other words, we want `assert` to be a method and `asserts` to be a struct/namespace).

**Review by tetromino — CHANGES_REQUESTED** [review]

File: lib/unittest.bzl (hunk 2 above)
Comment (kkress) [reply lib/unittest.bzl:668]: I can't do this.  assert is a keyword in starlark and fails. For now I've renamed to 'equals' to mirror the function in 'asserts'

I'm open to other suggestions.

**Comment by kkress** [comment]
Failures appear to be unrelated to my changes.

**Comment by tetromino** [comment]
For the failures - see https://github.com/bazelbuild/bazel-skylib/pull/349 (new buildifier release lints for more things). Maybe you can review?

**Review by tetromino — APPROVED** [review]
Thanks!

**Pull request merged**